File: InputTestCode/Kotlin/SwiftTour/2_ControlFlow.kt
// Iterating Array
val individualScores = arrayOf(75, 43, 103, 87, 12)
var teamScore = 0

fun iterateArray() {
    for (score in individualScores) {
        if (score > 50) {
            teamScore += 3
        } else {
            teamScore += 1
        }
    }
}

// Iterating Dictionary
val interestingNumbers = mapOf(
    "Prime" to arrayOf(2, 3, 5, 7, 11, 13),
    "Fibonacci" to arrayOf(1, 1, 2, 3, 5, 8),
    "Square" to arrayOf(1, 4, 9, 16, 25)
)
var largest = 0

fun iterateDictionary() {
    for ((key, numbers) in interestingNumbers) {
        for (number in numbers) {
            if (number > largest) {
                largest = number
            }
        }
    }
}

fun iterateDictionaryPlaceholder() {
    for ((_, numbers) in interestingNumbers) {
        for (number in numbers) {
            if (number > largest) {
                largest = number
            }
        }
    }
}

// While
var n = 2
fun useWhile() {
    while (n < 100) {
        n *= 2
    }
}

var m = 2
fun useRepeat() {
    do {
        m *= 2
    } while (m < 100)
}

// Range
var total = 0

fun useRange() {
    for (i in 0..(4 - 1)) {
        total += i
    }
}

// Optional values
fun checkOptionalValue(optionalString: String?): Boolean {
    return optionalString == null
}

// Optional in if
fun checkOptionalValueInIf(optionalName: String?): String {
    var greeting = "Hello!"

    optionalName?.let {
        greeting = "Hello, ${it}"
    }

    return greeting
}

// Default value for optional values
fun checkDefaultValue(nickname: String?): String {
    val fullName: String = "John Appleseed"
    val informalGreeting = "Hi ${nickname ?: fullName}"

    return informalGreeting
}

// Switch statements
fun switchStatement(vegetable: String): String {
    return when {
        vegetable == "celery" -> "Add some raisins and make ants on a log."
        vegetable == "cucumber" || vegetable == "watercress" -> "That would make a good tea sandwich."
        vegetable.endsWith("pepper") -> "Is it a spicy $vegetable?"
        else -> "Everything tastes good in soup."
    }
}

File: InputTestCode/Swift/SwiftTour/2_ControlFlow.swift
// Iterating Array
let individualScores = [75, 43, 103, 87, 12]
var teamScore = 0

func iterateArray() {
    for score in individualScores {
        if score > 50 {
            teamScore += 3
        } else {
            teamScore += 1
        }
    }
}

// Iterating Dictionary
let interestingNumbers = [
    "Prime": [2, 3, 5, 7, 11, 13],
    "Fibonacci": [1, 1, 2, 3, 5, 8],
    "Square": [1, 4, 9, 16, 25],
]
var largest = 0

func iterateDictionary() {
    for (key, numbers) in interestingNumbers {
        for number in numbers {
            if number > largest {
                largest = number
            }
        }
    }
}

func iterateDictionaryPlaceholder() {
    for (_, numbers) in interestingNumbers {
        for number in numbers {
            if number > largest {
                largest = number
            }
        }
    }
}

// While
var n = 2
func useWhile() {
    while n < 100 {
        n *= 2
    }
}

var m = 2
func useRepeat() {
    repeat {
        m *= 2
    } while m < 100
}

// Range
var total = 0
func useRange() {
    for i in 0..<4 {
        total += i
    }
}

// Optional values
func checkOptionalValue(optionalString: String?) -> Bool {
    return optionalString == nil
}

// Optional in if
func checkOptionalValueInIf(optionalName: String?) -> String {
    var greeting = "Hello!"

    if let name = optionalName {
        greeting = "Hello, \(name)"
    }

    return greeting
}

// Default value for optional values
func checkDefaultValue(nickname: String?) -> String {
    let fullName: String = "John Appleseed"
    let informalGreeting = "Hi \(nickname ?? fullName)"

    return informalGreeting
}

// Switch statements
func switchStatement(vegetable: String) -> String {
    switch vegetable {
    case "celery":
        return "Add some raisins and make ants on a log."
    case "cucumber", "watercress": 
        return "That would make a good tea sandwich."
    case let x where x.hasSuffix("pepper"):
        return "Is it a spicy \(x)?"
    default:
        return "Everything tastes good in soup."
    }
}

